cpp/src/hash/concurrent_unordered_map.cuh

```diff
@@ -422,16 +425,16 @@ class concurrent_unordered_map {
       if (cudaSuccess == status &&
           isPtrManaged(hashtbl_values_ptr_attributes)) {
         int dev_id = 0;
-        CUDA_RT_CALL(cudaGetDevice(&dev_id));
-        CUDA_RT_CALL(cudaMemPrefetchAsync(
+        CUDA_TRY(cudaGetDevice(&dev_id));
+        CUDA_TRY(cudaMemPrefetchAsync(
             m_hashtbl_values, m_capacity * sizeof(value_type), dev_id, stream));
       }
     }
 
     init_hashtbl<<<((m_capacity - 1) / block_size) + 1, block_size, 0, stream>>>(
         m_hashtbl_values, m_capacity, m_unused_key, m_unused_element);
-    CUDA_RT_CALL(cudaGetLastError());
-    CUDA_RT_CALL(cudaStreamSynchronize(stream));
+    CUDA_TRY(cudaGetLastError());
+    CUDA_TRY(cudaStreamSynchronize(stream));
```

Comment:
Is the synchronize required here? As long as the hash table is later used in the same stream as it is initialized on it shouldn't be, right? @nsakharnykh @jrhemstad thoughts?

Reply:
The synchronization is not required here and should be removed. In fact, in some other uses of the hash maps, I remember I had to remove it and keep a separate modified copy of cuDF. It’s a better design to let users synchronize outside the class if necessary.

Reply:
OK, I will remove the cudaStreamSynchronize calls.

Reply:
It seems like just removing cudaStreamSynchronize is not a proper resolution. If I am removing synchronization here, I should add something elsewhere or need to check whether all the use cases are properly using stream.

For example, we are taking stream in the function below,

https://github.com/rapidsai/cudf/blob/f8c1c0d5e5336d2890ca1a6f2ec9bedad5d4694d/cpp/src/groupby/hash/legacy/groupby.cu#L53

but, we are not passing stream to the create function here (which calls cudaStreamSynchronize at the end)

https://github.com/rapidsai/cudf/blob/f8c1c0d5e5336d2890ca1a6f2ec9bedad5d4694d/cpp/src/groupby/hash/legacy/groupby.cu#L92

but, using stream when using map returned by the create function.

https://github.com/rapidsai/cudf/blob/f8c1c0d5e5336d2890ca1a6f2ec9bedad5d4694d/cpp/src/groupby/hash/legacy/groupby.cu#L101

I will dig deeper to check what else I need to do to safely remove synchronization, here.

And we're still using legacy code, so I assume we cannot legacy code yet, but please let me know if I am doing something unnecessary.

Reply:
The port of groupby will use the hash maps.  It is not ported yet. I think the groupby port should fix this issue @jrhemstad

Reply:
OK, I will revert the last two commits expecting this issue to be addressed elsewhere.

Reply:
Oh, I misunderstood -- I thought you were reporting a problem. I didn't know you had already fixed it. Why revert it?

Reply:
OK, re-applied the changes.